examples/with-next/app/layout.tsx

```diff
@@ -0,0 +1,29 @@
+import type { Metadata } from 'next'
+import { Inter } from 'next/font/google'
+import { MSWProvider } from './msw-provider'
+
+if (process.env.NEXT_RUNTIME === 'nodejs') {
+  const { server } = require('../mocks/node')
+  server.listen()
+}
```

Comment:
is it recommended to do it here, or in the instrumentation file?

Reply:
@bitttttten, it is recommended to enable it in `layout.tsx`. You can forget about the instrumentation file. I've tried it before, but it doesn't support HMR and isn't a part of your app's life-cycle. That was a bad choice, and the Next.js team members have pointed that out for me.

Reply:
In what way is it suggested to change the mock per test if  the suggested way to set up the mocks is via the layout file?

Reply:
@votemike wouldnt each url be in a way idempotent? do you mean you want to call 1 url multiple times and see different results?
if you want to run different things depending on search parameters and what not you can do that in the mock

Reply:
Yes, the same URL with different return values.
Something like a product page.
Product A with stock
Product A out of stock
Product A coming soon
Product A 404 or 500ing.

I suppose I could vary the URL to pick up the mock I want. But in reality, production code would be calling the same URL and getting different return values at different times. Hence my desire to be able to vary the mock per test.

Reply:
I think I've sorted it: https://dev.to/votemike/server-side-mocking-for-playwright-in-nextjs-app-router-using-mock-service-worker-2p4i

Reply:
@votemike in your handler for that route you already have the one response mocked for the product. you could put that in an array along with the other results you want in that array. then randomly return a different response.